src/core/blueprint/index.rs

```diff
@@ -66,6 +66,37 @@ impl Index {
     pub fn get_mutation(&self) -> Option<&str> {
         self.schema.mutation.as_deref()
     }
+
+    pub fn is_type_implements(&self, type_name: &str, implements: &str) -> bool {
+        if type_name == implements {
+            return true;
+        }
+
+        if !matches!(
+            self.map.get(implements),
+            Some((Definition::Interface(_), _))
+        ) {
+            return false;
+        }
+
+        self.is_type_implements_rec(type_name, implements)
+    }
+
+    fn is_type_implements_rec(&self, type_name: &str, implements: &str) -> bool {
```

Comment:
I see a potential stackoverflow scenario. Can use a hashset to ensure to handle it. Also please add tests for the stack overflow condition also.

Reply:
I've checked the graphql spec again https://spec.graphql.org/October2021/#sec-Interfaces.Interfaces-Implementing-Interfaces and according to it the type should implement all transitive interfaces. So I removed the recursion check and left the only interface present in the list of `obj.implements`